File: src/main/java/fr/marstech/mtlinkspray/entity/LinkItem.java
package fr.marstech.mtlinkspray.entity;

import lombok.*;
import lombok.experimental.Accessors;
import lombok.experimental.SuperBuilder;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.util.Pair;

import java.time.LocalDateTime;
import java.util.Set;

@EqualsAndHashCode(callSuper = true)
@Accessors(chain = true)
@Data
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder
@Document(collection = "mt-link-spray-items")
public class LinkItem extends StandardEntity {

    @Id
    private String id;

    @NonNull
    private LinkItemTarget target;

    private String name;
    private Set<String> urlsToReplaceByJs;
    private Set<String> stringsToReplaceByJs;
    private Pair<LocalDateTime, String> expirationDateAndDestination;
}

File: src/main/java/fr/marstech/mtlinkspray/entity/LinkItem.kt
package fr.marstech.mtlinkspray.entity

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import org.springframework.data.mongodb.core.mapping.Field
import org.springframework.data.util.Pair
import java.time.LocalDateTime
import java.util.*

@Document(collection = "mt-link-spray-items")
data class LinkItem(
    @Id override val id: String = UUID.randomUUID().toString(),
    override val creationDate: LocalDateTime = LocalDateTime.now(),
    override val expiresAt: LocalDateTime? = null,
    override var isEnabled: Boolean = true,
    override var description: String? = null,
    override var metadata: MutableMap<String, String> = mutableMapOf(),
    override var author: HistoryItem,
    override var historyItems: MutableList<HistoryItem> = mutableListOf(),
    val target: LinkItemTarget,
) : StandardEntity